### PR TITLE
Implementation of TimeSlot:endHour parameter, check hour correction methods and check TimeSlots overlapping method

### PR DESCRIPTION
**Pull request by st46567**
Hello,
In this pull request I have done following changes in the class "TimeSlot.java":

- Add a paramater "endHour" instead of parameter duration
- Changed parameter name from "hour" to "startHour"
- Add constants "MIN_HOUR" and "MAX_HOUR" to determinate accepted interval for "startHour" and "endHour"
- Add methods "isStartHourCorrect" and "isEndHourCorrect" to check if hour parameters are in accepted interval => these methods are used in a constructor
- Add method "isOverlappingWithOtherTimeslot" to check if two TimeSlots are overlapping

- new tests in the test class "TimeSlotTest.java"

Can you check it please?

Thank you,
Jan Houžvička

### CODE REVIEW

File: src/main/java/cz/upce/fei/inptp/zz/entity/TimeSlot.java [hunk 1]
```diff
@@ -3,37 +3,63 @@
...
-    
+
     private Day day;
-    private int hour;
-    
-    // TODO: is this clear?
-    private int duration;
-    
-    public TimeSlot(Day day, int hour, int duration) {
+    private static final int MIN_HOUR = 8;
```
Comment (bLuma) [inline src/main/java/cz/upce/fei/inptp/zz/entity/TimeSlot.java]: Do we really need to impose hardcoded limit on hours?

File: src/main/java/cz/upce/fei/inptp/zz/entity/TimeSlot.java [hunk 2]
```diff
@@ -3,37 +3,63 @@
...
+    private static final int MAX_HOUR = 20;
+    private int startHour = MIN_HOUR;
+    private int endHour = MAX_HOUR;
+
+    public TimeSlot(Day day, int startHour, int endHour) throws IllegalArgumentException {
         this.day = day;
-        this.hour = hour;
-        this.duration = duration;
+
+        if (isStartHourCorrect(startHour)) {
```
Comment (bLuma) [inline src/main/java/cz/upce/fei/inptp/zz/entity/TimeSlot.java]: I personally prefer style

if (!check())
  throw ...
if (!check())
  throw 
...
(normal flow continues here)

File: src/main/java/cz/upce/fei/inptp/zz/entity/TimeSlot.java [hunk 3]
```diff
@@ -3,37 +3,63 @@
...
-    private int hour;
-    
-    // TODO: is this clear?
-    private int duration;
-    
-    public TimeSlot(Day day, int hour, int duration) {
+    private static final int MIN_HOUR = 8;
+    private static final int MAX_HOUR = 20;
+    private int startHour = MIN_HOUR;
+    private int endHour = MAX_HOUR;
```
Comment (bLuma) [inline src/main/java/cz/upce/fei/inptp/zz/entity/TimeSlot.java]: From startHour, endHour is not clear if both limits are considered as inclusive, exclusive or some combination...

File: src/main/java/cz/upce/fei/inptp/zz/entity/TimeSlot.java [hunk 4]
```diff
@@ -3,37 +3,63 @@
...
+    public int getStartHour() {
+        return startHour;
+    }
+
+    public int getEndHour() {
+        return endHour;
+    }
+
+    private boolean isStartHourCorrect(int startHour) {
+        return startHour < endHour && startHour >= MIN_HOUR && startHour < MAX_HOUR;
```
Comment (bLuma) [inline src/main/java/cz/upce/fei/inptp/zz/entity/TimeSlot.java]: I think that these checks are quite confusing... You're doing check using parameter startHour and also attribute endHour. You're mixing parameters and attributes and method name doesn't imply anything like this. Check for both parameters maybe would be better

File: src/test/java/cz/upce/fei/inptp/zz/entity/TimeSlotTest.java [hunk 5]
```diff
@@ -20,39 +20,93 @@ public void testSameSlotsShouldBeEqual() {
...
+        assertTrue(t_a.isOverlappingWithOtherTimeslot(t_b));
+    }
+
+    @Test
+    public void testTimeSlotsWithDifferentDayAreNotOverlapping() {
+        TimeSlot t_a = new TimeSlot(Day.Friday, 10, 13);
+        TimeSlot t_b = new TimeSlot(Day.Monday, 10, 13);
+
+        assertFalse(t_a.isOverlappingWithOtherTimeslot(t_b));
+    }
```
Comment (bLuma) [inline src/test/java/cz/upce/fei/inptp/zz/entity/TimeSlotTest.java:111]: Please add also checks for states:
`TA:---XXXX------------------`
`TB:--------------XXXXXX-----`
And also reversed TB-TA.

**Review by bLuma — CHANGES_REQUESTED** [review]
There are few issues to discuss and resolve, please see comments

**Comment by st46567** [comment]
Hello,
I tried to resolve issues described in the comments.

Can you check it please?
Thank you.

**Review by bLuma — APPROVED** [review]

**Pull request merged**